dsl_parser/functions.py

```diff
@@ -645,9 +645,28 @@ def _get_attribute_from_node_instance(ni, node, attribute_path, path, fn_name):
                                     path,
                                     raise_if_not_found=False,
                                     func_name=fn_name)
+        if isinstance(value, list):
+            value = _merge_function_args(value)
+
     return value
 
 
+def _merge_function_args(items):
+    for index, item in enumerate(items):
+        if is_function(item):
+            break
+    else:
```

Comment:
I really don't like the use of for/else here.
Also, doesn't this way of writing it result in us not processing more than one function?